tensorflow/core/kernels/mkl_conv_grad_input_ops.cc

```diff
@@ -0,0 +1,324 @@
+    dnnDelete_F32(mkl_convolutionbwdata);
+  }
+  void Compute(OpKernelContext* context) override {
+    const Tensor& input = MklGetInput(context, 0);
+    const Tensor& filter = MklGetInput(context, 1);
+
+    GetMklShape(context, 1, &(mkl_params.mkl_filter_shape));
+    bool filter_in_mkl_format = mkl_params.mkl_filter_shape.IsMklTensor();
+
+    const Tensor& out_backprop = MklGetInput(context, 2);
+    GetMklShape(context, 2, &(mkl_params.mkl_outback_shape));
+    bool outback_in_mkl_format = mkl_params.mkl_outback_shape.IsMklTensor();
+
+    TensorShape input_shape, filter_shape, outback_shape;
+
+    // Generate input shape.
+    OP_REQUIRES(
+        context, TensorShapeUtils::IsVector(input.shape()),
+        errors::InvalidArgument(
+            "Conv2DBackpropInput: input_sizes input must be 1-dim, not ",
+            input.dims()));
+    OP_REQUIRES_OK(
+        context, TensorShapeUtils::MakeShape(input.vec<int32>(), &input_shape));
+
+    // Generate shape for filter prop if input is in MKL format.
+    if (filter_in_mkl_format) {
+      OP_REQUIRES(context, mkl_params.mkl_filter_shape.GetDimension() == 4,
+                  errors::InvalidArgument(
+                      "Conv2DCustomBackpropInput: size must be 4-dim"));
+
+      MklSizesToTFSizes(context, data_format, mkl_params.mkl_filter_shape,
+                        &filter_shape);
+    } else {
+      filter_shape = filter.shape();
+    }
+
+    // Generate shape for outback prop if input is in MKL format.
+    if (outback_in_mkl_format) {
+      OP_REQUIRES(context, mkl_params.mkl_outback_shape.GetDimension() == 4,
+                  errors::InvalidArgument(
+                      "Conv2DCustomBackpropInput: size must be 4-dim"));
+
+      MklSizesToTFSizes(context, data_format, mkl_params.mkl_outback_shape,
+                        &outback_shape);
+    } else {
+      outback_shape = out_backprop.shape();
+    }
+
+    Conv2DBackpropDimensions dims;
+    OP_REQUIRES_OK(context,
+                   Conv2DBackpropComputeDimensions(
+                       "Conv2DCustomBackpropInput", input_shape, filter_shape,
+                       outback_shape, strides, padding, data_format, &dims));
+
+    int64 pad_top, pad_bottom;
+    int64 pad_left, pad_right;
+    OP_REQUIRES_OK(context, GetWindowedOutputSizeVerbose(
+                                dims.rows.input_size, dims.rows.filter_size,
+                                dims.rows.stride, padding,
+                                &dims.rows.output_size, &pad_top, &pad_bottom));
+    OP_REQUIRES_OK(context, GetWindowedOutputSizeVerbose(
+                                dims.cols.input_size, dims.cols.filter_size,
+                                dims.cols.stride, padding,
+                                &dims.cols.output_size, &pad_left, &pad_right));
+
+    mkl_params.in_dims = 4;
+
+    mkl_params.in_sizes[0] = static_cast<size_t>(dims.cols.input_size);
```

Comment:
Maybe break this up with line spaces between param types for readability:

    mkl_params.input_offset[0] = static_cast<int>(-pad_left);
￼    mkl_params.input_offset[1] = static_cast<int>(-pad_top);

￼    mkl_params.conv_strides[0] = static_cast<size_t>(dims.cols.stride);
￼    mkl_params.conv_strides[1] = static_cast<size_t>(dims.rows.stride);